libparsec/crates/types/tests/unit/manifest.rs

```diff
@@ -0,0 +1,256 @@
+    let expected_version = 1;
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &dummy_without_compression,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::Compression)
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &dummy,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::Serialization)
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &SecretKey::generate(),
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::Decryption)
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &alice.local_symkey,
+            &SigningKey::generate().verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::Signature)
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &expected_author,
+            now,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::UnexpectedAuthor {
+            expected: Box::new(expected_author),
+            got: Some(Box::new(alice.device_id.clone())),
+        })
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            expected_timestamp,
+            Some(id),
+            Some(0)
+        ),
+        Err(DataError::UnexpectedTimestamp {
+            expected: expected_timestamp,
+            got: now,
+        })
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(expected_id),
+            Some(0)
+        ),
+        Err(DataError::UnexpectedId {
+            expected: expected_id,
+            got: id,
+        })
+    );
+
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &data,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(expected_version)
+        ),
+        Err(DataError::UnexpectedVersion {
+            expected: expected_version,
+            got: 0,
+        })
+    );
```

Comment:
There is a lot of case tested here, I think adding a short comment per case could help understand what each case is doing :thinking:

WDYT ?

Reply:
Done